runestone/activecode/activecode.py

```diff
@@ -229,6 +232,60 @@ def run(self):
         if 'gradebutton' not in self.options:
             self.options['gradebutton'] = ''
 
+        if self.content:
+            if '====' in self.content:
+                idx = self.content.index('====')
+                source = "\n".join(self.content[:idx])
+                suffix = "\n".join(self.content[idx+1:])
+            else:
+                source = "\n".join(self.content)
+                suffix = "\n"
+        else:
+            source = '\n'
+            suffix = '\n'
+        try:
+            engine = create_engine(env.config['dburl'])
+            meta = MetaData()
+            course_name = env.config['course_id']
```

Comment:
course_id should already be available from the template arguments in pavement.py, we should not duplicate this bit of information in conf.py.